shared/src/main/scala/squants/electro/AreaElectricChargeDensity.scala

```diff
@@ -0,0 +1,49 @@
+package squants.electro
+
+import squants.space.Meters
+import squants.{AbstractQuantityNumeric, Area, Dimension, Length, PrimaryUnit, Quantity, SiUnit, UnitConverter, UnitOfMeasure}
+
+/**
+  *
+  * @author Nicolas Vinuesa
+  * @since 1.4
+  *
+  * @param value Double
+  */
+final class AreaElectricChargeDensity private (val value: Double, val unit: AreaElectricChargeDensityUnit)
+  extends Quantity[AreaElectricChargeDensity] {
+
+  def dimension = AreaElectricChargeDensity
+
+  def *(that: Area): ElectricCharge = Coulombs(this.toCoulombsSquareMeters * that.toSquareMeters)
+  def *(that: Length): LinearElectricChargeDensity = CoulombsMeters(this.toCoulombsSquareMeters * that.toMeters)
+
+  def toCoulombsSquareMeters = to(CoulombsSquareMeters)
+}
+
+object AreaElectricChargeDensity extends Dimension[AreaElectricChargeDensity] {
+  private[electro] def apply[A](n: A, unit: AreaElectricChargeDensityUnit)(implicit num: Numeric[A]) = new AreaElectricChargeDensity(num.toDouble(n), unit)
+  def apply = parse _
+  def name = "AreaElectricChargeDensity"
+  def primaryUnit = CoulombsSquareMeters
+  def siUnit = CoulombsSquareMeters
+  def units = Set(CoulombsSquareMeters)
+}
+
+trait AreaElectricChargeDensityUnit extends UnitOfMeasure[AreaElectricChargeDensity] with UnitConverter {
+  def apply[A](n: A)(implicit num: Numeric[A]) = AreaElectricChargeDensity(n, this)
+}
+
+object CoulombsSquareMeters extends AreaElectricChargeDensityUnit with PrimaryUnit with SiUnit {
```

Comment:
I wonder if this would be better named as `CoulombsPerSquareMeter` ? I don't have a strong opinion about it.